plugins/prefect/access_key.go

```diff
@@ -0,0 +1,86 @@
+package prefect
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func AccessKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.AccessKey,
+		DocsURL:       sdk.URL("https://docs.prefect.io/2.10.13/cloud/users/api-keys/"),
+		ManagementURL: sdk.URL("https://app.prefect.cloud/my/api-keys"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "Key used to authenticate to Prefect.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 40,
+					Prefix: "pnu_",
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.URL,
+				MarkdownDescription: "URL for the Prefect workspace.",
+				Secret:              false,
+				Composition: &schema.ValueComposition{
+					Length: 124,
+					Prefix: "https://api.prefect.cloud/api/accounts/",
+					Charset: schema.Charset{
+						Uppercase: false,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			TryPrefectConfigFile(),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"PREFECT_API_KEY": fieldname.APIKey,
+	"PREFECT_API_URL": fieldname.URL,
+}
+
+// TODO: Check if the platform stores the Access Key in a local config file, and if so,
```

Comment:
I don't quite know if prefect stores the access key in a local config file. Would be great if I can get some help in understanding this more

Reply:
@avinash-santhanagopalan You can run `find ~/.* -maxdepth 3 -path "*prefect*"` to check if there's a config file on your computer. Depending on what you find, you can use one of the helper functions (ToString, ToYAML, ToJSON, etc) in `sdk/importer/file_importer.go` to parse it.

Reply:
So prefect creates a `.prefect` directory in the user's `$HOME` directory. There is a `profiles.toml` file in this directory that looks like this

```toml
active = "default"

[profiles.default]
PREFECT_API_KEY = <APIKey>
PREFECT_API_URL = <URL>
```

Reply:
From a Shell Plugins point of view, it makes sense to store all the envvars that are necessary for the connection. From the state of the PR so far, I assume it is just `PREFECT_API_KEY` and `PREFECT_API_URL` that are minimum-required for the authentication process? If so, I suggest we start with these two fields and expand as we go.

Let me know if you have any questions about the TOML importer!

Reply:
I looked into this TOML importer but I'm not sure if prefect would fetch the credentials from this file to authenticate itself. I think they also have plans to drop support for creating this `.prefect` directory so can we pursue this as a follow up and just restrict this first iteration to the current changes?

Reply:
Indeed the settings in the tool file seem not to be relevant for our autentication workflow. Let's just skip the file importing for now, and only do env var importing.

Reply:
@avinash-santhanagopalan